receiver/awslambdareceiver/receiver.go

```diff
@@ -4,23 +4,266 @@
 	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awslambdareceiver/internal"
+)
+
+type eventType string
+
+const (
+	s3Event           eventType = "S3Event"
+	cwEvent           eventType = "CloudWatchEvent"
+	customReplayEvent eventType = "replayFailedEvents"
+
+	// logDeployedTrigger define the string key to convey the deployed trigger type in logs
+	logDeployedTrigger = "deployedTrigger"
+	// logInvokedTrigger define the string key to convey the invoked trigger(derived by event content) type in logs
+	logInvokedTrigger = "invokedTrigger"
 )
 
-type awsLambdaReceiver struct{}
+var (
+	errEncoderNotFound                  = errors.New("extension not found")
+	_                  receiver.Metrics = (*awsLambdaReceiver)(nil)
+	_                  receiver.Logs    = (*awsLambdaReceiver)(nil)
+)
+
+type awsLambdaReceiver struct {
+	cfg        *Config
+	logger     *zap.Logger
+	buildInfo  component.BuildInfo
+	s3Provider internal.S3Provider
+	newHandler func(context.Context, component.Host, internal.S3Provider) (lambdaEventHandler, error)
+
+	handler lambdaEventHandler
+}
+
+func newLogsReceiver(cfg *Config, set receiver.Settings, next consumer.Logs) (receiver.Logs, error) {
+	return &awsLambdaReceiver{
+		cfg:        cfg,
+		logger:     set.Logger,
+		buildInfo:  set.BuildInfo,
+		s3Provider: &internal.S3ServiceProvider{},
+		newHandler: func(
+			ctx context.Context,
+			host component.Host,
+			s3Provider internal.S3Provider,
+		) (lambdaEventHandler, error) {
+			return newLogsHandler(
+				ctx, cfg, set, host, next,
+				s3Provider,
+			)
+		},
+	}, nil
+}
 
-func newAWSLambdaReceiver(_ *Config, _ *receiver.Settings, _ consumer.Logs) *awsLambdaReceiver {
-	return &awsLambdaReceiver{}
+func newMetricsReceiver(cfg *Config, set receiver.Settings, next consumer.Metrics) (receiver.Metrics, error) {
+	if cfg.S3Encoding == "" {
+		return nil, errors.New("s3_encoding is required for metrics")
+	}
+	return &awsLambdaReceiver{
+		cfg:        cfg,
+		logger:     set.Logger,
+		buildInfo:  set.BuildInfo,
+		s3Provider: &internal.S3ServiceProvider{},
+		newHandler: func(
+			ctx context.Context,
+			host component.Host,
+			s3Provider internal.S3Provider,
+		) (lambdaEventHandler, error) {
+			return newMetricsHandler(ctx, cfg, set, host, next, s3Provider)
+		},
+	}, nil
 }
 
-func (*awsLambdaReceiver) Start(_ context.Context, _ component.Host) error {
+// Start registers the main handler function for
+// when lambda is triggered
+func (a *awsLambdaReceiver) Start(ctx context.Context, host component.Host) error {
+	handler, err := a.newHandler(ctx, host, a.s3Provider)
+	if err != nil {
+		return fmt.Errorf("failed to create the lambda event handler: %w", err)
+	}
+	a.handler = handler
+
+	// Only start Lambda runtime if we're actually in a Lambda environment
+	// This prevents errors during tests when Lambda environment variables are not set
```

Comment:
Can we arrange for the tests to set the environment variables instead? I think that Start should return with an error if someone mistakenly tries to use the receiver outside of a Lambda environment.